### DISCUSSION TITLE
Window.ExtendsContentIntoTitleBar vs AppWindowTitleBar.ExtendsContentIntoTitleBar

### DISCUSSION THREAD

**Discussion by adamplonka** [microsoft/WindowsAppSDK#2816] [General]
Both `Window` and `AppWindowTitleBar` have an `ExtendsContentIntoTitleBar` property but they have different behavior.
Which one is preferred and should be used?

`Window.ExtendsContentIntoTitleBar` seems like an obvious choice because it's "natively" supported by Window without any fishy Microsoft.UI.Win32Interop calls, but it behaves differently than` AppWindowTitleBar` one - seems like it's using XAML ChromeWindow and looks worse, flickers on window resize and the "restore" icon is blurred. However, the XAML is more customizable and maybe it's supported on older windows systems too?
The AppWindowTitleBar is used in all samples on https://docs.microsoft.com/en-us/windows/apps/develop/title-bar?tabs=wasdk#tabpanel_5_wasdk site
but it requires windows 11 to work.

Is `AppWindowTitleBar.IsCustomizationSupported()` check required when using `Window.ExtendsContentIntoTitleBar`?
Are there any plans on supporting title bar customization on windows 10 in the future?

**Comment by AndrewKeepCoding** [comment]
Title bar customization is very confusing (for the moment?)...🙄

> Which one is preferred and should be used?

I wouldn't use the ``AppWindowTitleBar`` yet since it only supports **Win11** but it totally depends on you case of course.
<br>
> However, the XAML is more customizable and maybe it's supported on older windows systems too?

Yes.
<br>
> Are there any plans on supporting title bar customization on windows 10 in the future?

I've heard (in one of the Community Calls I think?) that the team is considering to improve title bar customization.

**Reply by ghost1372** [reply]
version 1.2-preview1 now supports win10
